packages/react-components/src/components/table-output-component.tsx

```diff
@@ -216,20 +226,30 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
 
     private async handleTimeSelectionChange(range: TimelineChart.TimeGraphRange) {
         const start = Math.trunc(this.props.range.getstart() + range.start);
-        if (this.lastIndex.timestamp === start) {
+        const end = Math.trunc(this.props.range.getstart() + range.end);
+        if (this.lastIndex.startTimestamp === start && this.lastIndex.endTimestamp === end) {
             return;
         }
-        const index = await this.fetchTableIndex(start);
-        if (index && this.gridApi) {
-            this.lastIndex = { timestamp: start, index: index };
+        const startIndex = await this.fetchTableIndex(start);
+        const endIndex = await this.fetchTableIndex(end);
+        if (startIndex && endIndex && this.gridApi) {
+            this.lastIndex = {
+                startTimestamp: start,
+                endTimestamp: end,
+                startIndex: startIndex,
+                endIndex: endIndex
+            };
             this.gridApi.deselectAll();
-            this.gridApi.ensureIndexVisible(index);
-            const node = this.gridApi.getDisplayedRowAtIndex(index);
-            if (node.id) {
-                node.setSelected(true);
-            } else {
-                this.pendingIndex = index;
+            for (let index = startIndex; index <= endIndex; index++) {
```

Comment:
This does not work when the selection is right-to-left (e.g. Shift+left) and endIndex < startIndex.